core/src/main/java/org/mini2dx/Tappybird/TappyBirdGame.java

```diff
@@ -1,33 +1,150 @@
+
+    CollisionBox[] collisionRectanglesBottom, collisionRectanglesTop;
+
+    int pillarIndexHead, pillarIndexTail;
+    int pillarTiming;
 	
 	@Override
     public void initialise() {
-    	texture = new Texture("mini2Dx.png");
+	    GRAVITY = 0.0f;
+	    flyingspeed = 7.5f;
+
+	    pillars = new Pillars[MAX_PILLARS];
+	    collisionRectanglesBottom = new CollisionBox[MAX_PILLARS];
+	    collisionRectanglesTop = new CollisionBox[MAX_PILLARS];
+
+    	inputHandler = new InputHandler();
+    	player = new Player();
+    	ground1 = new TopBottomEdge();
+    	ground2 = new TopBottomEdge();
+        ground2.generateHazardAtPos(gameWidth,gameHeight - ground1.getGroundTextureHeight());
+    	background1 = new Background();
+        background2 = new Background();
+        background2.generateHazardAtPos(gameWidth,0.0f);
+        fitViewport = new FitViewport(gameWidth, gameHeight);
+
+        pillarTiming = 60;
     }
     
     @Override
     public void update(float delta) {
 
+        pillarTiming -= new Random().nextInt(2)+1;
+
+	    if(inputHandler.SpacePressed())
+	        GRAVITY = 0.4f;
+
+        if (pillarTiming < 0 ){
+            pillarTiming = 60;
+            pillars[pillarIndexTail] = new Pillars();
+            pillars[pillarIndexTail].generateHazardAtPos(gameWidth + 100f, randomFloatMinMax(-100f,0f), randomFloatMinMax(125f,200f));
+            collisionRectanglesBottom[pillarIndexTail] = pillars[pillarIndexTail].collisionRectBottom;
+            collisionRectanglesTop[pillarIndexTail] = pillars[pillarIndexTail].collisionRectTop;
+            pillarIndexTail = (pillarIndexTail + 1) % MAX_PILLARS;
+        }
+
+        for(int i=0; i<MAX_PILLARS; i++) {
+            if (pillars[i] != null) {
+                pillars[i].update();
+                if (pillars[pillarIndexHead]!= null && pillars[pillarIndexHead].getHazardXPos()<-50){
+                    pillars[pillarIndexHead] = null;
+                    collisionRectanglesBottom[pillarIndexHead] = null;
+                    collisionRectanglesTop[pillarIndexHead] = null;
+                    pillarIndexHead = (pillarIndexHead + 1 ) % MAX_PILLARS;
+                }
+            }
+        }
+
+
+        player.update(inputHandler.SpacePressed());
+        ground1.update();
+        ground2.update();
+        background1.update();
+        background2.update();
+
+        if (player.getPlayerYPos()<0 || player.getPlayerYPos()>gameHeight-player.getPlayerTextureHeight()){
+            initialise();
+        }
+
+        checkCollisions();
+
+        //TODO Code here to exit the game.
+        if (inputHandler.EscPressed()){
+
+        }
     }
     
     @Override
     public void interpolate(float alpha) {
-    
+        ground1.interpolate(alpha);
+        ground2.interpolate(alpha);
     }
     
     @Override
     public void render(Graphics g) {
-		g.drawTexture(texture, 0f, 0f);
+	    fitViewport.apply(g);
+
+	    background1.render(g);
+        background2.render(g);
```

Comment:
Try to keep indentation consistent (IntelliJ can format this automatically via Code -> Reformat Code)